MovieChecker/MovieChecker/MovieList.m

```diff
@@ -0,0 +1,81 @@
+
+@interface MovieList ()
+
+@property (nonatomic, strong) NSURLSession *session;
+
+@end
+
+@implementation MovieList
+
+- (void)fetchMovies
+{
+    NSURL *url = [RottenTomatoesHelperMethods boxOfficeURL];
+    NSURLSessionDataTask *dataTask = [self.session
+                                      dataTaskWithURL:url
+                                      completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
+                                          if (error) {
+                                              // TODO: Handle the error!
+                                              return;
+                                          }
+                                          NSHTTPURLResponse *httpResp = (NSHTTPURLResponse *)response;
+                                          if (httpResp.statusCode != 200) {
+                                              // TODO: Handle the bad status code!
+                                              return;
+                                          }
+                                          self.movies = [RottenTomatoesHelperMethods interpretBoxOfficeMoviesFromData:data
+                                                                                                withkey:@"movies"];
+                                          [[NSOperationQueue mainQueue] addOperationWithBlock:^{
+                                              [self.delegate synchMoviesWithArray:self.movies];
+                                          }];
+                                      }];
+    [dataTask resume];
+}
+
+- (void)downloadImageForMovie:(Movie *)movie
+{
+    NSURL *url = [NSURL URLWithString:movie.imagePath];
+    NSURLSessionDataTask *downloadImageTask = [[NSURLSession sharedSession]
+                                               dataTaskWithURL:url
+                                               completionHandler:^(NSData *data,
+                                                                   NSURLResponse *response,
+                                                                   NSError *error) {
+                                                   if (error) {
+                                                       // TODO: Handle the error!
+                                                       return;
+                                                   }
+                                                   NSHTTPURLResponse *httpResp = (NSHTTPURLResponse *)response;
+                                                   if (httpResp.statusCode != 200) {
+                                                       // TODO: Handle the bad status code!
+                                                       return;
+                                                   }
+                                                   movie.image = [[UIImage alloc] initWithData:data];
+                                               }];
+    [downloadImageTask resume];
```

Comment:
It appears that the only purpose of the local variable `downloadImageTask` is to call `resume`—consider reformatting as:
```objc
    [[[NSURLSession sharedSession]
      dataTaskWithURL:url
      completionHandler:^(NSData *data,
                          NSURLResponse *response,
                          NSError *error) {
          if (error) {
              // TODO: Handle the error!
              return;
          }
          NSHTTPURLResponse *httpResp = (NSHTTPURLResponse *)response;
          if (httpResp.statusCode != 200) {
              // TODO: Handle the bad status code!
              return;
          }
          movie.image = [[UIImage alloc] initWithData:data];
      }]
     resume];
```